addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2065,6 +2068,22 @@ var SnippetsMenu = Widget.extend({
         cacheSnippetTemplate[this.options.snippets] = this._defLoadSnippets;
         return this._defLoadSnippets;
     },
+    /**
+     * Visually hide or display this snippet menu
+     * @param {boolean} foldState
+     */
+    setFolded: function (foldState = true) {
+        if (foldState) {
+            this.el.classList.add('d-none');
+            this.el.ownerDocument.body.classList.remove('editor_has_snippets');
+        } else {
+            this.el.classList.remove('d-none');
+            this.el.ownerDocument.body.classList.add('editor_has_snippets');
+        }
```

Comment:
In general, I would suggest this form:
```suggestion
        this.el.classList.toggle('d-none', foldState);
        this.el.ownerDocument.body.classList.toggle('editor_has_snippets', !foldState);
```